server/stream.go

```diff
@@ -1806,18 +1808,21 @@ func (mset *stream) updateWithAdvisory(config *StreamConfig, sendAdvisory bool)
 					}
 
 					mset.sources[s.iname] = si
-					mset.setStartingSequenceForSource(s.iname, s.External)
-					mset.setSourceConsumer(s.iname, si.sseq+1, time.Time{})
+					needsStartingSeqNum[s.iname] = struct{}{}
 				} else {
 					// source already exists
 					delete(currentIName, s.iname)
 				}
 			}
-			// What is left in cuurentIName needs to be deleted.
+			// What is left in curentIName needs to be deleted.
 			for iName := range currentIName {
 				mset.cancelSourceConsumer(iName)
 				delete(mset.sources, iName)
```

Comment:
Should we delete from needsStartingSeqNum here?

Reply:
No need, it's not in needsStartingSeqNum. currentIName is initially populated with all the sources and here we delete if not needed, while needsStartingSeqNum starts empty and gets added only if needed.